nuxeo-coldstorage-package/src/main/resources/install/templates/coldstorage/nxserver/config/nuxeo-coldstorage-config.xml.nxftl

```diff
@@ -2,19 +2,20 @@
 <component name="org.nuxeo.ecm.core.coldstorage.ci.test">
     <extension target="org.nuxeo.ecm.core.blob.BlobManager" point="configuration">
         <blobprovider name="glacier">
-            <class>org.nuxeo.ecm.blob.s3.S3BlobProvider</class>
+            <class>${nuxeo.core.binarymanager}</class>
             <property name="awsid">${nuxeo.aws.accessKeyId}</property>
             <property name="awssecret">${nuxeo.aws.secretKey}</property>
             <property name="region">${nuxeo.aws.region}</property>
-            <property name="bucket">${nuxeo.glacier.bucket}</property>
-            <property name="bucket_prefix">${nuxeo.glacier.bucket_prefix}/</property>
+            <property name="bucket">${nuxeo.aws.glacier.bucket}</property>
+            <property name="bucket_prefix">${nuxeo.aws.glacier.bucket_prefix}/</property>
             <property name="directdownload">true</property>
             <property name="directdownload.expire">3600</property>
-            <property name="cachesize">100MB</property>
-            <property name="connection.max">50</property>
-            <property name="connection.retry">3</property>
-            <property name="connection.timeout">50000</property>
-            <property name="socket.timeout">50000</property>
+            <property name="cachesize">${nuxeo.s3storage.cachesize}</property>
+            <property name="connection.max">${nuxeo.s3storage.connection.max}</property>
+            <property name="connection.retry">${nuxeo.s3storage.connection.retry}</property>
+            <property name="connection.timeout">${nuxeo.s3storage.connection.timeout}</property>
+            <property name="socket.timeout">${nuxeo.s3storage.socket.timeout}</property>
+            <property name="keyStrategy">managed</property>
```

Comment:
Perhaps, I missed it, but It think it will be worth to extract it as variable, even on platform / S3 Direct upload / Aspera the property is hardcode. It will give us more flexibility if there something to fix.

Reply:
my proposal 
`nuxeo.s3storage.key.strategy=managed`

Reply:
No, for direct upload with the new code since [NXP-30044](https://jira.nuxeo.com/browse/NXP-30044) the key strategy should not be made configurable anymore, and is fixed to "managed". Changing it would break direct upload optimizations.

Reply:
yep, my idea is to give more flexibility in a case where we have to change things, and the to put `managed ` in the extracted variable. But we will hardcoded it and avoid the usage of the property.

I have a question regarding that, do we have to put the `nuxeo.core.blobstore.digestAsync=true` as we will do a move from bucket to another (S3 with standard class to S3 with glacier class). I mean adding the configuration on our new provider [blobprovider name="glacier"](https://github.com/nuxeo/nuxeo-coldstorage/pull/31/files#diff-9284fa7ee7eb373af7d983bf6ff0715d6dd6dc49f0d77f172e4aafdec239a952L4)

I am wondering about that, as Aspera addon-on we did `nuxeo.core.blobstore.digestAsync=true` but on S3 Direct it's not the case.

Reply:
`nuxeo.core.blobstore.digestAsync=true` is more efficient but has the side effect that digests are not immediately available after direct upload is finished. There is a behavior change. So the fact that it's not enabled by default in the standard config is due to backward compatibility reasons.
I don't know if it should be enabled or not by default for Glacier (I would think not), but it's a product decision.

Reply:
Thanks, @efge
We will check with the product.
Meanwhile, I think it will be worth adding to our Glacier Blob provider contribution.